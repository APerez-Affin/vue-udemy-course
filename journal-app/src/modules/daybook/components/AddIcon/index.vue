<template>
  <button class="btn btn-primary" :class="icon" @click="$emit('on:click')">
    <font-awesome-icon :icon="['fas', icon]" />
  </button>
</template>

<script>
export default {
  name: "AddIcon",
  props: {
    icon: {
      type: String,
      default: "plus",
    },
  },
};
</script>

<style lang="scss" scoped>
@import "./AddIcon.scss";
</style>