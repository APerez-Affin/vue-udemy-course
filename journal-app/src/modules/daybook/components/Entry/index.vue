<template>
  <div class="entry" @click="$router.push({ name: 'entry', params: { id: 10} })">
    <div class="entry__title">
      <span class="entry__day">15</span>
      <div class="entry__date">
        <span class="entry__month"> Julio</span>
        <span class="entry__year">2021, jueves</span>
      </div>
    </div>
    <div class="entry__description">
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Voluptates
      fugiat exercitationem at quo, harum eum magni illum numquam iure
      blanditiis ut, assumenda animi molestias porro adipisci odit magnam.
      Deserunt, natus.
    </div>
  </div>
</template>

<script>
export default {
  name: "Entry",
};
</script>

<style lang="scss" scoped>
@import "./Entry.scss";
</style>