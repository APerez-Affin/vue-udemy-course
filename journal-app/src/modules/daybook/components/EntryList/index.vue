<template>
  <div class="entrylist">
    <div class="entrylist__search">
      <input
        type="text"
        class="input input__text"
        placeholder="Buscar entrada"
        v-model.trim="term"
      />
    </div>
    <div class="entrylist__scrollarea">
      <Entry
        v-for="entry of entriesByTerm"
        :key="entry.id"
        :entry="entry"
      />
    </div>
  </div>
</template>

<script>
import { defineAsyncComponent } from "vue";
import { mapGetters } from "vuex";
export default {
  name: "EntryList",
  components: {
    Entry: defineAsyncComponent(() => import("../../components/Entry")),
  },
  computed: {
    ...mapGetters("journal", ["getEntriesByTerm"]),
    entriesByTerm() {
      return this.getEntriesByTerm(this.term);
    },
  },
  data() {
    return {
      term: "",
    };
  },
};
</script>

<style lang="scss" scoped>
@import "./EntryList.scss";
</style>