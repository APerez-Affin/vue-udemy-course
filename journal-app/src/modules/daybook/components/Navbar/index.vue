<template>
  <nav class="navbar">
      <div class="navbar__content">
        <a class="navbar__brand">
          <figure class="navbar__brand-container">
            <img
              alt="Vue logo"
              class="navbar__brand-logo"
              src="@/assets/logo.png"
            />
          </figure>
          <span class="navbar__brand-logoName">Daybook</span>
        </a>

        <div class="navbar__actions">
          <button class="btn btn--outline">
            <FontAwesomeIcon icon="sign-out-alt" />
          </button>
        </div>
      </div>
  </nav>
</template>

<script>
export default {
  name: "NavBar",
};
</script>

<style lang="scss" scoped>
@import "./Navbar.scss";
</style>