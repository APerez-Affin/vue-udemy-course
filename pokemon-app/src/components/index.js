export { default as PokemonPicture } from "./PokemonPicture";
export { default as PokemonOptions } from "./PokemonOptions";
