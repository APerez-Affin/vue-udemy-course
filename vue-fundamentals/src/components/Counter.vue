<template>
  <h2>{{ customTitle }}</h2>
  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
  <p data-testid="counter">{{ counter }}</p>
  <div class="buttons">
    <button data-testid="btn-increase" @click="increase">+1</button>
    <button data-testid="btn-decrease" @click="decrease">-1</button>
  </div>
</template>

<script>
  export default {
    props: {
      title: {
        type: String,
      },
      start: {
        type: Number,
        default: 100,

        validator(value) {
          return value >= 0;
        },
      },
    },
    name: "Counter",
    data() {
      return {
        counter: this.start,
      };
    },
    computed: {
      squareCounter() {
        return this.counter * this.counter;
      },
      customTitle() {
        return this.title || "Counter";
      },
    },
    methods: {
      increase() {
        this.counter++;
      },
      decrease() {
        this.counter--;
      },
    },
  };
</script>

<style>
  button {
    background-color: #64b687;
    border-radius: 8px;
    color: white;
    cursor: pointer;
    margin: 0 8px;
    padding: 8px 16px;
    transition: 0.3 ease-in-out;
    border: none;
    outline: none;
  }
  button:hover {
    background-color: #5aaa;
  }
</style>
