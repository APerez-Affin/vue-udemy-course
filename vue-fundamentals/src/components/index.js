export { default as Counter } from "./Counter";
export { default as Indecision } from "./Indecision";
