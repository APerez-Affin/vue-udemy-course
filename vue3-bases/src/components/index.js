export { default as InitialComponent } from '@/components/InitialComponent';