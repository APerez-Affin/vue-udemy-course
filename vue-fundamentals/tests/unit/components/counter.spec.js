import { shallowMount, mount } from "@vue/test-utils";
import { Counter } from "@/components/index";

/*
 * Shallowmount: Monta el componente de una forma más ligera y menos intrusiva.
 * Mount: Monta el componente de una forma más intrusiva accediendo a los hijos.
 */

describe("Counter Component", () => {
  let wrapper;

  // Carga el componente antes de ejecutar los test
  beforeEach(() => {
    wrapper = shallowMount(Counter);
  });

  test("Should match with the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should be greater than 10 ", () => {
    /* Espera que sea mayor que 10 */
    expect(20).toBeGreaterThan(10);
  });

  test("Should display the title Counter by default ", () => {
    /* Busca la etiqueta */
    const h = wrapper.find("h2");

    /* Verifica si existe la etiqueta */
    expect(h.exists()).toBeTruthy();
    /* Verifica si el valor por defecto es counter */
    expect(h.text()).toBe("Counter");
  });

  test("Should display the value 100 on the p", () => {
    /* Seleccionamos el testid */
    const value = wrapper.find('[data-testid="counter"]');

    /* Verificamos si el valor del p es 100 */
    expect(value.text()).toBe("100");
  });

  test("Should increase the counter value plus 1 ", async () => {
    const increase = wrapper.find('[data-testid="btn-increase"]');
    const value = wrapper.find('[data-testid="counter"]');

    /* Simula un click asincrono */
    await increase.trigger("click");
    expect(value.text()).toBe("101");
  });

  test("Should decrease the counter value less 1 ", async () => {
    const decrease = wrapper.find('[data-testid="btn-decrease"]');
    const value = wrapper.find('[data-testid="counter"]');

    /* Simula un click asincrono */
    await decrease.trigger("click");
    expect(value.text()).toBe("99");
  });

  test("Should assign a default value ", () => {
    /* Accedemos a las props */
    const { start } = wrapper.props();
    const value = wrapper.find('[data-testid="counter"]').text();

    expect(Number(value)).toBe(start);
  });

  test("Should show the prop title ", () => {
    /* Enviamos un texto de ejemplo a la prop */
    const wrapper = shallowMount(Counter, {
      props: {
        title: "Texto de ejemplo"
      }
    });

    const value = wrapper.find("h2").text();

    expect(value).toBe('Texto de ejemplo');
  });

});
