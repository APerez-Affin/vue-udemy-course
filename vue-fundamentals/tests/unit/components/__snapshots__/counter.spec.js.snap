// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Should math with the snapshot 1`] = `
<h2>Counter</h2>
<p>100 <sup>2</sup> = 10000</p>
<p data-testid="counter">100</p>
<div class="buttons"><button data-testid="btn-increase">+1</button><button data-testid="btn-decrease">-1</button></div>
`;
